### DISCUSSION TITLE
'const { walletProvider } = useAppKitProvider() '   walletProvider is undefined

### DISCUSSION THREAD

**Discussion by bentong-chain** [WalletConnect/walletconnect-monorepo#5379] [Developer Support]
<script setup lang="ts">
  import { useAppKitProvider, useAppKitAccount } from "@reown/appkit/vue";
  import { BrowserProvider, Contract, formatUnits } from 'ethers'

  const USDTAddress = '0x617f3112bf5397D0467D315cC709EF968D9ba546'

  // The ERC-20 Contract ABI, which is a common contract interface
  // for tokens (this is the Human-Readable ABI format)
  const USDTAbi = [
    'function name() view returns (string)',
    'function symbol() view returns (string)',
    'function balanceOf(address) view returns (uint)',
    'function transfer(address to, uint amount)',
    'event Transfer(address indexed from, address indexed to, uint amount)'
  ]

  function Components() {
    const { address, isConnected } = useAppKitAccount()
    const { walletProvider } = useAppKitProvider()

    async function getBalance() {
    if (!isConnected) throw Error('User disconnected')

    const ethersProvider = new BrowserProvider(walletProvider)
    const signer = await ethersProvider.getSigner()
    // The Contract object
    const USDTContract = new Contract(USDTAddress, USDTAbi, signer)
    const USDTBalance = await USDTContract.balanceOf(address)

    console.log(formatUnits(USDTBalance, 18))
  }
  return <button onClick={getBalance}>Get User Balance</button>
}
</script>

'const { walletProvider } = useAppKitProvider() '   walletProvider is undefined

**Comment by bentong-chain** [comment]
ethers v6

**Reply by ggranile** [reply]
did you find a solution?

**Reply by thirteensuits** [reply]
yes has anyone found a soltuon to this?

**Comment by AwanFvnky** [comment]
Maybe a little down

**Comment by Favvie** [comment]
It will work if you specify the chainNamespace:

const { walletProvider } = useAppKitProvider("eip155")

**Reply by DobromirKirovLime** [reply]
which is not specified in the docs...

**Reply by thirteensuits** [reply]
Hello, I am also getting the undefined for walletProvider.
using MetaMask for BNB Chain
successfully connects and can pass wallet address, but walletProvider is undefined

**Reply by DobromirKirovLime** [reply]
@thirteensuits found that the wallet provider will be defined when you connect to some specific wallet, pretty inconvenient that this is not mentioned in the docs...

**Reply by thirteensuits** [reply]
that's great!

can you be more specific? were you using Metamask or operating on EVM?

**Reply by DobromirKirovLime** [reply]
with metamask

**Reply by thirteensuits** [reply]
so is it:

const { walletProvider } = useAppKitProvider("eip155")

or

const { walletProvider } = useAppKitProvider("metamask" or walletID or something)

**Reply by DobromirKirovLime** [reply]
`const { walletProvider } = useAppKitProvider("eip155")` 
 later in your code you call the modal and connect to some wallet and after that the `walletProvider` should be populated

**Reply by harissatti** [reply]
does any one know how we can wallet provider? because when we do this in code other issues are rises,so we get walletprovider in proper way?

**Comment by harissatti** [comment]
does any one know how we can wallet provider? because when we do this in code other issues are rises,so we get walletprovider in proper way?

**Comment by harissatti** [comment]
const { walletProvider } = useAppKitProvider("eip155")
because this doesnot a proper solution.need a proper solution.because it rise other issues

**Reply by DobromirKirovLime** [reply]
well you need to ask the maintainers of this repo, for the proper fix
